src/components/MlEphantConversation2.tsx

```diff
@@ -19,25 +23,78 @@ import type {
 } from '@src/machines/mlEphantManagerMachine2'
 import type { ReactNode } from 'react'
 import { Fragment, useEffect, useRef, useState } from 'react'
+import {
+  DEFAULT_ML_COPILOT_MODEL,
+  DEFAULT_ML_COPILOT_REASONING_EFFORT,
+} from '@src/lib/constants'
 
 export interface MlEphantConversationProps {
   isLoading: boolean
   conversation?: Conversation
   contexts: MlEphantManagerPromptContext[]
   billingContext: BillingContext
-  onProcess: (request: string, forcedTools: Set<MlCopilotTool>) => void
+  onProcess: (
+    request: string,
+    model: MlCopilotSupportedModels,
+    reasoningEffort: MlReasoningEffort,
+    forcedTools: Set<MlCopilotTool>
+  ) => void
   disabled?: boolean
   hasPromptCompleted: boolean
   userAvatarSrc?: string
   defaultPrompt?: string
 }
 
+const ML_REASONING_EFFORT: Readonly<MlReasoningEffort[]> = Object.freeze([
+  'low',
+  // 'medium',
+  'high',
+])
+const ML_COPILOT_SUPPORTED_MODELS: Readonly<MlCopilotSupportedModels[]> =
+  Object.freeze(['gpt5_nano', 'gpt5_mini', 'gpt5_codex', 'gpt5', 'o3'])
 const ML_COPILOT_TOOLS: Readonly<MlCopilotTool[]> = Object.freeze([
   'edit_kcl_code',
   'text_to_cad',
   'mechanical_knowledge_base',
   'web_search',
 ])
+const ML_REASONING_EFFORT_META = Object.freeze({
+  low: {
+    pretty: 'Fast',
+    icon: (props: { className: string }) => (
+      <CustomIcon name="stopwatch" className={props.className} />
+    ),
+  },
+  medium: {
+    pretty: 'Thoughtful',
+    icon: (props: { className: string }) => (
+      <CustomIcon name="questionMark" className={props.className} />
```

Comment:
```suggestion
      <CustomIcon name="brain" className={props.className} />
```

Could you use the brain icon here, or did you not like it?

Reply:
I tweaked the viewBox 😬